File: MoneyUtils.py
class MoneyUtils:
    MONEY_UNIT = " MGA"

    @staticmethod
    def is_amount_valid(amount, balance=None, max_transaction=None):
        if amount < 0:
            print("Amount must be > 0", MoneyUtils.MONEY_UNIT)
            return False

        if balance is not None and amount > balance:
            print("Amount must be <= Balance:", balance, MoneyUtils.MONEY_UNIT)
            return False

        if max_transaction is not None and amount > max_transaction:
            print("Amount must be <= Max Transaction:", max_transaction, MoneyUtils.MONEY_UNIT)
            return False

        return True

File: MoneyUtils.cs
namespace WalletApp
{
    public static class MoneyUtils
    {
        public const string MoneyUnit = " MGA";

        public static bool IsAmountValid(double amount)
        {
            if (amount < 0)
            {
                Console.WriteLine("Amount must be > 0" + MoneyUnit);
                return false;
            }
            return true;
        }

        public static bool IsAmountValid(double amount, double balance)
        {
            if (!IsAmountValid(amount))
            {
                return false;
            }

            if (amount > balance)
            {
                Console.WriteLine("Amount must be <= Balance: " + balance + MoneyUnit);
                return false;
            }
            return true;
        }

        public static bool IsAmountValid(double amount, double balance, double maxTransaction)
        {
            if (!IsAmountValid(amount, balance))
            {
                return false;
            }

            if (amount > maxTransaction)
            {
                Console.WriteLine("Amount must be <= Max Transaction: " + maxTransaction + MoneyUnit);
                return false;
            }
            return true;
        }
    }
}
